src/app/core/services/label.service.ts

```diff
@@ -70,36 +95,92 @@ export class LabelService {
   }
 
   /**
-   * Stores the json data from Github api into the list of arrays in this service
-   * @param labels: the json data of the label
+   * Returns an array of Preset Labels.
    */
-  private populateLabelLists(labels: Array<{}>): void {
+  private getRequiredLabels(): Label[] {
+    const requiredLabels: Label[] = [];
 
-    for (const label of labels) {
-      // Get the name and color of each label and store them into the service's array list
-      const labelName = String(label['name']).split('.');
-      const labelType = labelName[0];
-      const labelValue = labelName[1];
-      const labelColor = String(label['color']);
-
-      // Check for duplicate labels
-      if (this.labelColorMap.has(labelValue)) {
-        continue;
+    for (const category of Object.keys(this.REQUIRED_LABELS)) {
+      for (const labels of Object.values(this.REQUIRED_LABELS[category])) {
+        requiredLabels.push(labels as Label);
       }
+    }
+
+    return requiredLabels;
+  }
+
+  /**
+   * Stores the json data from Github api into the list of arrays in this service
```

Comment:
Um... this method stores the given set of labels irrespective of whether they came from Github api. Don't explain a function based on how it is currently used (because you cannot stop someone else using it in a different way); explain what it _does_ (that is the part you can be sure of).

Reply:
Sure, I will rework the method description.